src/compile/facet.ts

```diff
@@ -242,6 +233,13 @@ export class FacetModel extends ModelWithField {
     );
   }
 
+  private columnDistinctSignal() {
+    // In facetNode.assemble(), the name is always this.getName('column') + '-layout'.
```

Comment:
I wonder whether we should use output nodes instead so that we can merge if two subtrees are equivalent. I guess it's not relevant for now since you didn't add an aggregate node but maybe for the future.